keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -0,0 +1,453 @@
+
+```
+	// AllocatedResources represents the compute resources allocated for this container by the
+	// node. Kubelet sets this value to Container.Resources.Requests upon successful pod admission
+	// and after successfully admitting desired pod resize.
+	// +featureGate=InPlacePodVerticalScaling
+	// +optional
+	AllocatedResources ResourceList `json:"allocatedResources,omitempty" protobuf:"bytes,10,rep,name=allocatedResources,casttype=ResourceList,castkey=ResourceName"`
+
+	// Resources represents the compute resource requests and limits that have been successfully
+	// enacted on the running container after it has been started or has been successfully resized.
+	// +featureGate=InPlacePodVerticalScaling
+	// +optional
+	Resources *ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,11,opt,name=resources"`
+```
+
+One field reflects the resource requests and limits and the other actual allocated resources.
+
+This structure will contain standard resources as well as extended resources. As noted in the comment: https://github.com/kubernetes/kubernetes/pull/124227#issuecomment-2130503713, it is only logical to also include the status of those allocated resources. 
+
+The proposal is to keep this structure as-is to simplify parsing of well-known ResourceList data type by various consumers. Typical scenario would be to compare if the `AllocatedResources` match the desired state.
+
+The proposal is to introduce an additional field:
+
+```
+// AllocatedResourcesStatus represents the status of various resources
+// allocated for this Pod.
+AllocatedResourcesStatus ResourcesStatus 
+
+type ResourcesStatus map[ResourceName]ResourceStatus
+
+type ResourceStatus struct {
+    // map of unique Resource ID to its health.
+    // At a minimum, ResourceID must uniquely identify the Resource
+    // allocated to the Pod on the Node for the lifetime of a Pod.
+    // See ResourceID type for it's definition.
+    Resources map[ResourceID] ResourceHealth
+    
+    // allow to extend this struct in future with the overall health fields or things like Device Plugin version
+}
+
+// ResourceID is calculated based on the source of this resource health information.
+// For DevicePlugin:
+//   deviceplugin:DeviceID, where DeviceID is from the Device structure of DevicePlugin's ListAndWatchResponse type: https://github.com/kubernetes/kubernetes/blob/eda1c780543a27c078450e2f17d674471e00f494/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto#L61-L73
+// DevicePlugin ID is usually a constant for the lifetime of a Node and typically can be used to uniquely identify the device on the node.
+// For DRA:
+//   dra:<driver name>[/<pool name>]/<device name>: such a device can be looked up in the information published by that DRA driver to learn more about it. It is designed to be globally unique in a cluster.
```

Comment:
```suggestion
//   dra:<driver name>/<pool name>/<device name>: such a device can be looked up in the information published by that DRA driver to learn more about it. It is designed to be globally unique in a cluster.
```

In the latest KEP revision, we no longer allow an empty pool name.

I know I proposed it, but I am not sure whether such a string format will pass API review. The alternative is a separate "type" field which replaces the prefix. Can be clarified during API review of the actual implementation.

You also need to consider how much validation you want to add for this.

I can put a `ValidateDRADeviceID` function into some shared package for this.